core/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -196,7 +196,13 @@ public long paramAsLong(String key, long defaultValue) {
 
     @Override
     public boolean paramAsBoolean(String key, boolean defaultValue) {
-        return Booleans.parseBoolean(param(key), defaultValue);
+        String rawParam = param(key);
+        // treat the sheer presence of a parameter as "true"
```

Comment:
Maybe "Treat empty string as `true` because that allows the presence of the url parameter to mean "turn this on"".